src/taskpane/pages/dashboard.tsx

```diff
@@ -53,24 +54,44 @@ function onCheckboxChange(ev: React.FormEvent<HTMLElement | HTMLInputElement>) {
 }
 
 function unCheckCheckbox(item: bib): bib {
-  return { ...item, isSelected: false };
+  return {
+    ...item,
+    isSelected: false,
+    label: null,
+    locator: null,
+    prefix: null,
+    suffix: null,
+  };
 }
 
 function Dashboard({ citeSupport }: DashboardProps): ReactElement {
-  const originalItems = data.map((item) => ({ ...item, isSelected: false }));
-  const [items, setItems] = useState(originalItems);
+  const originalItems = data.map((item) => ({
+    ...item,
+    label: "",
+    locator: "",
+    suffix: "",
+    prefix: "",
+    isSelected: false,
+  }));
+  const [items, setItems] = useState<Array<bib>>(originalItems);
   const [citationItemsIDs, _setCitationItemsIDs] = useState([]);
   const [isCitationSelected, setIsCitationSelection] = useState(false);
   const itemsIDsInSelectedCitation = useRef(citationItemsIDs);
-  const setCitationItemsIDs = (ids: Array<string>) => {
-    itemsIDsInSelectedCitation.current = ids;
-    _setCitationItemsIDs(ids);
+  const setCitationItemsIDs = (itemsMetadata: Array<CitationItem>) => {
+    itemsIDsInSelectedCitation.current = itemsMetadata;
+    _setCitationItemsIDs(itemsMetadata);
   };
 
   const checkedItems = items
     .filter((item) => item.isSelected)
-    .map((item) => {
-      return { id: item.id };
+    .map((item: bib) => {
+      return {
```

Comment:
This is the format for `citationItems`. We can not pass `Array<bib>` to citeproc.

Reply:
Why not? The "duct typing" nature of type/javascript allows to pass different objects as long they have the required properties. Since you are not renaming anything here, I think it would just work if remove the map here.

Moreover, the dashboard shouldn't worry about the format citeproc expects. For transforming the data/properties in the correct format for CSL, we have the citation worker. So even if the map turns out to be necessary, I would move it there.